cli/src/main/java/org/eclipse/hono/cli/AbstractCliClient.java

```diff
@@ -44,32 +54,91 @@
      */
     protected Context ctx;
     /**
-     * The Spring Boot profiles that are active.
+     * The class containing all client properties.
      */
-    protected List<String> activeProfiles;
-
+    protected ClientConfigProperties honoClientConfig;
+    /**
+     * To connection created.
+     */
+    protected HonoConnection con;
+    /**
+     * To stop the executions of internal commands.
+     */
+    protected CountDownLatch connLatch;
     /**
-     * Sets the Spring environment.
-     * 
-     * @param env The environment.
-     * @throws NullPointerException if environment is {@code null}.
+     * Spring Shell inputReader.
      */
     @Autowired
-    public final void setActiveProfiles(final Environment env) {
-        Objects.requireNonNull(env);
-        activeProfiles = Arrays.asList(env.getActiveProfiles());
+    protected @Lazy InputReader inputReader;
+    /**
+     * Spring Shell output helper.
+     * <p>
+     * it can be passed to the method class instead to use the log (it's more stylish).
+     */
+    @Autowired
+    protected ShellHelper shellHelper;
+    /**
+     * The client factory methods.
+     */
+    protected ConnectionLifecycle<HonoConnection> clientFactory;
```

Comment:
IMHO we should parameterize the `AbstractCliClient` class with the type of client factory that it supports.
That way we would also get rid of the ugly cast that is done in the subclasses when a command is being executed.

Reply:
Any reason why you do not want to parameterize the `AbstractCliClient` class with the client factory type?